### ISSUE TITLE
Incompatibility with latest spatialdata-plot

### ISSUE THREAD

**Issue by grst** [marcovarrone/cellcharter#33]
### Report

Hi, 

I can't import cellcharter in an environment that has the latest spatialdata-plot package installed. It seems a private method that's used by cellcharter got removed.

```pytb
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
Cell In[2], line 2
      1 import squidpy as sq
----> 2 import cellcharter as cc
      3 import scanpy as sc
      4 from lightning.pytorch import seed_everything

File /cfs/sturmgre/conda/envs/1368-0008_visium2/lib/python3.10/site-packages/cellcharter/__init__.py:3
      1 from importlib.metadata import version
----> 3 from . import datasets, gr, pl, tl
      5 __all__ = ["gr", "pl", "tl", "datasets"]
      7 __version__ = version("cellcharter")

File /cfs/sturmgre/conda/envs/1368-0008_visium2/lib/python3.10/site-packages/cellcharter/pl/__init__.py:4
      2 from ._group import enrichment, proportion
      3 from ._nhood import diff_nhood_enrichment, nhood_enrichment
----> 4 from ._shape import plot_boundaries, plot_shape_metrics

File /cfs/sturmgre/conda/envs/1368-0008_visium2/lib/python3.10/site-packages/cellcharter/pl/_shape.py:15
     13 from anndata import AnnData
     14 from scipy.stats import ttest_ind
---> 15 from spatialdata_plot.pl.utils import _get_colors_for_categorical_obs
     17 from ._utils import adjust_box_widths
     20 def plot_boundaries(
     21     adata: AnnData,
     22     sample: str,
   (...)
     26     show_cells: bool = True,
     27 ) -> None:

File /cfs/sturmgre/conda/envs/1368-0008_visium2/lib/python3.10/site-packages/spatialdata_plot/__init__.py:3
      1 from importlib.metadata import version
----> 3 from . import pl, pp
      5 __all__ = ["pl", "pp"]
      7 __version__ = version("spatialdata-plot")

File /cfs/sturmgre/conda/envs/1368-0008_visium2/lib/python3.10/site-packages/spatialdata_plot/pl/__init__.py:1
----> 1 from .basic import PlotAccessor
      3 __all__ = [
      4     "PlotAccessor",
      5 ]

File /cfs/sturmgre/conda/envs/1368-0008_visium2/lib/python3.10/site-packages/spatialdata_plot/pl/basic.py:26
     23 from spatialdata._core.data_extent import get_extent
     25 from spatialdata_plot._accessor import register_spatial_data_accessor
---> 26 from spatialdata_plot.pl.render import (
     27     _render_images,
     28     _render_labels,
     29     _render_points,
     30     _render_shapes,
     31 )
     32 from spatialdata_plot.pl.render_params import (
     33     CmapParams,
     34     ImageRenderParams,
   (...)
     40     _FontWeight,
     41 )
     42 from spatialdata_plot.pl.utils import (
     43     _get_cs_contents,
     44     _get_elements_to_be_rendered,
   (...)
     55     save_fig,
     56 )

File /cfs/sturmgre/conda/envs/1368-0008_visium2/lib/python3.10/site-packages/spatialdata_plot/pl/render.py:21
     19 from scanpy._settings import settings as sc_settings
     20 from spatialdata._core.data_extent import get_extent
---> 21 from spatialdata.models import PointsModel, get_table_keys
     22 from spatialdata.transformations import (
     23     get_transformation,
     24 )
     26 from spatialdata_plot._logging import logger

ImportError: cannot import name 'get_table_keys' from 'spatialdata.models' (/cfs/sturmgre/conda/envs/1368-0008_visium2/lib/python3.10/site-packages/spatialdata/models/__init__.py)
```

Downgrading spatialdata-plot to 0.0.6 solved the issue. 

### Version information

absl-py==2.1.0
affine==2.4.0
aiobotocore==2.5.4
aiohttp==3.9.5
aioitertools==0.11.0
aiosignal==1.3.1
alabaster==0.7.16
anndata==0.10.7
anyio==4.3.0
argon2-cffi==23.1.0
argon2-cffi-bindings==21.2.0
array_api_compat==1.6
arrow==1.3.0
asciitree==0.3.3
asttokens==2.4.1
async-lru==2.0.4
async-timeout==4.0.3
attrs==23.2.0
Babel==2.14.0
beautifulsoup4==4.12.3
bleach==6.1.0
botocore==1.31.17
cell2location @ git+https://github.com/BayraktarLab/cell2location.git@b2f38944dd13f3b3024e10f97abaf3240f6cccfe
cellcharter==0.2.0
certifi==2024.2.2
cffi==1.16.0
charset-normalizer==3.3.2
chex==0.1.86
click==8.1.7
click-plugins==1.1.1
cligj==0.7.2
cloudpickle==3.0.0
colorcet==3.1.0
comm==0.2.2
contextlib2==21.6.0
contourpy==1.2.1
cycler==0.12.1
dask==2024.4.2
dask-expr==1.0.14
dask-image==2023.8.1
datashader==0.16.1
debugpy==1.8.1
decorator==5.1.1
defusedxml==0.7.1
distributed==2024.4.2
docrep==0.3.2
docutils==0.21.2
etils==1.7.0
exceptiongroup==1.2.1
executing==2.0.1
fasteners==0.19
fastjsonschema==2.19.1
fiona==1.9.6
flax==0.8.3
fonttools==4.51.0
fqdn==1.5.1
frozenlist==1.4.1
fsspec==2023.6.0
geopandas==0.14.4
greenlet==3.0.3
h11==0.14.0
h5py==3.11.0
httpcore==1.0.5
httpx==0.27.0
idna==3.7
igraph==0.11.4
imageio==2.34.1
imagesize==1.4.1
importlib_metadata==7.1.0
importlib_resources==6.4.0
inflect==7.2.1
ipykernel==6.29.4
ipython==8.24.0
isoduration==20.11.0
jax==0.4.26
jaxlib==0.4.26
jedi==0.19.1
Jinja2==3.1.3
jmespath==1.0.1
joblib==1.4.0
json5==0.9.25
jsonpointer==2.4
jsonschema==4.22.0
jsonschema-specifications==2023.12.1
jupyter-cache==1.0.0
jupyter-events==0.10.0
jupyter-lsp==2.2.5
jupyter_client==8.6.1
jupyter_core==5.7.2
jupyter_server==2.14.0
jupyter_server_terminals==0.5.3
jupyterlab==4.1.8
jupyterlab_pygments==0.3.0
jupyterlab_server==2.27.1
kiwisolver==1.4.5
lazy_loader==0.4
legacy-api-wrap==1.4
leidenalg==0.10.2
lightkit==0.5.0
lightning==2.1.4
lightning-utilities==0.11.2
llvmlite==0.42.0
locket==1.0.0
markdown-it-py==3.0.0
MarkupSafe==2.1.5
matplotlib==3.8.4
matplotlib-inline==0.1.7
matplotlib-scalebar==0.8.1
mdit-py-plugins==0.4.0
mdurl==0.1.2
mistune==3.0.2
ml-collections==0.1.1
ml-dtypes==0.4.0
more-itertools==10.2.0
msgpack==1.0.8
mudata==0.2.3
multidict==6.0.5
multipledispatch==1.0.0
multiscale_spatial_image==0.11.2
myst-nb==1.1.0
myst-parser==3.0.1
natsort==8.4.0
nbclient==0.10.0
nbconvert==7.16.4
nbformat==5.10.4
nest-asyncio==1.6.0
networkx==3.3
notebook==7.1.3
notebook_shim==0.2.4
numba==0.59.1
numcodecs==0.12.1
numpy==1.26.4
numpyro==0.14.0
ome-zarr==0.8.3
omnipath==1.0.8
opencv-python==4.9.0.80
opt-einsum==3.3.0
optax==0.2.2
orbax-checkpoint==0.5.10
overrides==7.7.0
packaging==24.0
pandas==2.2.2
pandocfilters==1.5.1
param==2.1.0
parso==0.8.4
partd==1.4.1
patsy==0.5.6
pexpect==4.9.0
pillow==10.3.0
PIMS==0.6.1
platformdirs==4.2.1
prometheus_client==0.20.0
prompt-toolkit==3.0.43
protobuf==5.26.1
psutil==5.9.8
ptyprocess==0.7.0
pure-eval==0.2.2
pyarrow==16.0.0
pycave==3.2.1
pycparser==2.22
pyct==0.5.0
pygeos==0.14
Pygments==2.17.2
pynndescent==0.5.12
pyparsing==3.1.2
pyproj==3.6.1
pyro-api==0.1.2
pyro-ppl==1.8.6
python-dateutil==2.9.0.post0
python-json-logger==2.0.7
pytorch-lightning==1.9.5
pytz==2024.1
PyYAML==6.0.1
pyzmq==26.0.3
rasterio==1.3.10
referencing==0.35.1
requests==2.31.0
rfc3339-validator==0.1.4
rfc3986-validator==0.1.1
rich==13.7.1
rpds-py==0.18.0
s3fs==2023.6.0
scanpy==1.10.1
scikit-image==0.23.2
scikit-learn==1.4.2
scipy==1.13.0
scvi-tools==1.1.2
seaborn==0.13.2
Send2Trash==1.8.3
session-info==1.0.0
shapely==2.0.4
six==1.16.0
sknw==0.15
slicerator==1.1.0
sniffio==1.3.1
snowballstemmer==2.2.0
snuggs==1.4.7
sortedcontainers==2.4.0
soupsieve==2.5
spatial_image==0.3.0
spatialdata==0.0.15
spatialdata-plot==0.0.6
Sphinx==7.3.7
sphinx-copybutton==0.5.2
sphinxcontrib-applehelp==1.0.8
sphinxcontrib-devhelp==1.0.6
sphinxcontrib-htmlhelp==2.0.5
sphinxcontrib-jsmath==1.0.1
sphinxcontrib-qthelp==1.0.7
sphinxcontrib-serializinghtml==1.1.10
SQLAlchemy==2.0.29
squidpy==1.4.1
stack-data==0.6.3
statsmodels==0.14.2
stdlib-list==0.10.0
tabulate==0.9.0
tblib==3.0.0
tensorstore==0.1.58
terminado==0.18.1
texttable==1.7.0
threadpoolctl==3.5.0
tifffile==2024.4.24
tinycss2==1.3.0
tomli==2.0.1
toolz==0.12.1
torch==1.12.1
torchmetrics==0.11.4
tornado==6.4
tqdm==4.66.2
traitlets==5.14.3
typeguard==4.2.1
types-python-dateutil==2.9.0.20240316
typing_extensions==4.11.0
tzdata==2024.1
umap-learn==0.5.6
uri-template==1.3.0
urllib3==1.26.18
validators==0.28.1
wcwidth==0.2.13
webcolors==1.13
webencodings==0.5.1
websocket-client==1.8.0
wrapt==1.16.0
xarray==2023.12.0
xarray-dataclasses==1.7.0
xarray-datatree==0.0.14
xarray-schema==0.0.3
xarray-spatial==0.4.0
yarl==1.9.4
zarr==2.17.2
zict==3.0.0
zipp==3.18.1

**Comment by marcovarrone** [comment]
Hi @grst, I believe the problem could be solved by updating spatialdata to a version > 0.1.0.
It looks like `get_table_keys` was added in v0.1.0.
https://github.com/scverse/spatialdata/releases/tag/v0.1.0

Let me know if this solves the problem.

**Comment by marcovarrone** [comment]
Fixed by https://github.com/CSOgroup/cellcharter/commit/8bf08e7938ada45424811c4d8736d76a15fe2b9f
